### ISSUE TITLE
maximum recursion depth exceeded when using %(version)s with versions='timestamp'

### ISSUE THREAD

**Issue by nathan-hoad** [miracle2k/webassets#252]
>>> import webassets
    >>> b = webassets.Bundle('foo.js', output='asdf-%(version)s.js')
    >>> e = webassets.Environment('.', '', versions='timestamp')
    >>> e.register('js', b)
    <Bundle output=asdf-%(version)s.js, filters=(), contents=('foo.js',)>
    >>> try:
    ...     print e['js'].urls()
    ...     print 'success!'
    ... except RuntimeError as e:
    ...     print 'ouch', e
    ...
    ouch maximum recursion depth exceeded

The offending block of function calls:

      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/bundle.py", line 217, in get_version
        version = env.versions.determine_version(self, env)
      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/version.py", line 102, in determine_version
        return self.find_recent_most_timestamp(bundle, env)
      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/version.py", line 125, in find_recent_most_timestamp
        for filename in get_all_bundle_files(bundle, env):
      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/bundle.py", line 720, in get_all_bundle_files
        for _, c in bundle.resolve_contents(env):
      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/bundle.py", line 160, in resolve_contents
        result.remove(self.resolve_output(env))
      File "/home/nathan/.local/lib/python2.7/site-packages/webassets/bundle.py", line 239, in resolve_output
        output = output % {'version': version or self.get_version(env)}

So basically what's happening is a bundle is trying to determine its version for the filename (because of `%(version)s`) by resolving the output, which in turn needs the version, so we get back to trying to determine the version.

A quick fix is to handle `RuntimeError` in `Bundle.resolve_contents`, along with the already handled `ValueError` and `BundleError`. It's gross I know, but it works due to caching - I found this code quite confusing and was in need of a quick fix, and this is what worked.

**Comment by miracle2k** [comment]
Closing for age.

**Comment by nicolaspanel** [comment]
@nathan-hoad @miracle2k same probleme with `versions='timestamp'`.
 Have you found a workaround that do not require to edit `flask-assets`'s code ?
